# Market Demand Analysis

1. Data processing

In [1]:
import pandas as pd
import plotly.express as px

In [2]:
data = pd.read_csv('/content/Workforce_Skills_Gap_Analysis.csv')

**Skill Demand Forecast:** Analyze the 'Skills Most in Demand' data to predict future workforce needs and guide the creation of future training initiatives.

In [5]:
# First, we need to make a copy of the data
df = data.copy()

# Let's filter for most recent years (2022 and 2023) to predict future trends
df = df[df['Year'].isin([2022, 2023])]

# Next, we calculate the total number of employees needing reskilling by skill and sort descending
df = df.groupby('Skills Most in Demand')['Employees Needing Reskilling (Thousands)'].sum().sort_values(ascending=False)

# Then we reset the index to return a DataFrame
df = df.reset_index()

df

Skills Most in Demand  Employees Needing Reskilling (Thousands)
0  Sales and Customer Management                                      1484
1           Healthcare Expertise                                      1375
2                  Data Analysis                                       907
3                Cloud Computing                                       800
4             Project Management                                       652
5        AI and Machine Learning                                       444
6              Digital Marketing                                       428
7    Renewable Energy Technology                                       339
8                  Cybersecurity                                       232

In [6]:
fig = px.bar(df,
             x='Employees Needing Reskilling (Thousands)',
             y='Skills Most in Demand',
             color='Skills Most in Demand',
             title="Skill Demand Forecast",
             labels={'Employees Needing Reskilling (Thousands)': 'Employees (Thousands)',
                     'Skills Most in Demand': 'Skills'},
             template='plotly_dark')

fig.show()

The data indicates a strong future demand in the following areas:


Sales and Customer Management tops the list with 1484 thousand employees needing reskilling, suggesting a critical need for sales skills and customer handling capabilities.

Healthcare Expertise is the second most demanded skill with 1375 thousand employees requiring reskilling, indicating a growing need in the healthcare industry.

Data Analysis and Cloud Computing are also in high demand, with 907 and 800 thousand employees needing training, respectively, pointing towards the ongoing digital transformation in businesses.

Lower on the list but still significant, AI and Machine Learning, and Cybersecurity indicate emerging tech fields where future workforce needs are likely to increase, with 444 and 232 thousand employees needing reskilling respectively.

In [7]:
df = data.copy()

# Convert 'Year' to datetime format
df['Year'] = pd.to_datetime(df['Year'], format='%Y')

# Filter data up to the current year
df = df[df['Year'] <= '2024']

# Calculate the total cost of training for each industry and country
df['Total Training Cost (USD)'] = df['Employees Needing Reskilling (Thousands)'] * df['Average Training Cost Per Employee (USD)'] * 1000

# Group by 'Country' and 'Industry' and calculate the sum of 'Employees Needing Reskilling (Thousands)'
# and 'Total Training Cost (USD)' and the average of 'Business Impact of Skill Shortage (USD Million)'
df = df.groupby(['Country', 'Industry']).agg({'Employees Needing Reskilling (Thousands)': 'sum',
                                              'Total Training Cost (USD)': 'sum',
                                              'Business Impact of Skill Shortage (USD Million)': 'mean'}).reset_index()

# Calculate the cost per employee needing reskilling
df['Cost Per Employee Needing Reskilling (USD)'] = df['Total Training Cost (USD)'] / (df['Employees Needing Reskilling (Thousands)'] * 1000)

# Sort by 'Total Training Cost (USD)' in descending order
df = df.sort_values('Total Training Cost (USD)', ascending=False)

df

Country        Industry  Employees Needing Reskilling (Thousands)  \
24      India    Construction                                       578   
35         UK          Retail                                       483   
5      Brazil          Retail                                       554   
6      Brazil  Transportation                                       444   
23    Germany      Healthcare                                       847   
30      Japan          Retail                                       573   
13     Canada      Technology                                       643   
16      China       Education                                       475   
33         UK         Finance                                       787   
34         UK   Manufacturing                                       466   
37        USA   Manufacturing                                       335   
22     France      Technology                                       378   
29      Japan     Hospitality                                       530   
12     Canada          Retail                                       752   
21     France          Retail                                       398   
36        USA    Construction                                       486   
20     France     Hospitality                                       396   
9      Canada          Energy                                       331   
1   Australia         Finance                                       452   
26      Japan          Energy                                       424   
0   Australia       Education                                       346   
32         UK       Education                                       183   
14     Canada  Transportation                                       361   
3   Australia          Retail                                       218   
4      Brazil          Energy                                       284   
7      Canada    Construction                                       232   
28      Japan      Healthcare                                       239   
18      China          Retail                                       141   
15      China    Construction                                       139   
39        USA      Technology                                       176   
38        USA          Retail                                       142   
10     Canada         Finance                                       438   
8      Canada       Education                                       116   
17      China          Energy                                       359   
27      Japan         Finance                                       202   
31      Japan      Technology                                       117   
2   Australia   Manufacturing                                        78   
25      India       Education                                       100   
11     Canada      Healthcare                                       177   
19     France          Energy                                        59   

    Total Training Cost (USD)  \
24                 2384929000   
35                 2285556000   
5                  2237168000   
6                  2082360000   
23                 2015510000   
30                 1949886000   
13                 1868472000   
16                 1830175000   
33                 1742534000   
34                 1709288000   
37                 1565790000   
22                 1467177000   
29                 1332898000   
12                 1328000000   
21                 1323748000   
36                 1317546000   
20                 1299672000   
9                  1273357000   
1                  1075760000   
26                  875984000   
0                   872612000   
32                  780495000   
14                  747270000   
3                   719036000   
4                   718236000   
7                   695304000   
28                  682823000   
18    

2. Data visualization

In [8]:
fig = px.bar(df, x='Total Training Cost (USD)', y='Country', color='Industry',
             title='Total Training Cost by Industry and Country',
             labels={'Total Training Cost (USD)': 'Total Training Cost (USD)',
                     'Country': 'Country'},
             hover_data=['Employees Needing Reskilling (Thousands)',
                         'Business Impact of Skill Shortage (USD Million)',
                         'Cost Per Employee Needing Reskilling (USD)'],
             color_discrete_sequence=px.colors.qualitative.Bold,
             template='plotly_dark')

fig.show()

3. Insights generation

The analytics task provides insights into the reskilling needs and associated costs across various industries and countries.

* **Retail industry in the UK and Brazil** are facing significant reskilling needs with 483 thousand and 554 thousand employees respectively, costing over $2.2 billion each.

* **Germany's Healthcare sector needs** to reskill 847 thousand employees, with a total training cost exceeding $2 billion.

  Among countries, **India's Construction sector** has the highest demand for reskilling, with 578 thousand employees at a cost of about $2.4 billion.

* **The Technology sector in the USA** needs to reskill 176 thousand employees which would cost about $556 million, indicating a high cost per employee for reskilling in this industry.

These insights can guide resource allocation in training and development efforts.